test/tla/specs/checker/Solver.tla

```diff
@@ -0,0 +1,54 @@
+---------------------------- MODULE Solver ------------------------------------
+(*
+ * An abstract model of a constraint solver:
+ *
+ * 1. The states are represented with integers.
+ *
+ * 2. The constraints are represented with sets of integers (predicates) and
+ * sets of pairs of integers (relations).
+ *
+ * Igor Konnov, Informal Systems, 2022
+ *)
+
+EXTENDS Integers, Sequences, Solver_typedefs
+
+CONSTANTS
+    \* The set of all potential states.
+    \*
+    \* @type: Set(Int);
+    STATES
+
+(*
+ * context.unary contains constraints over states that are layered by time
+ * frames.  Every set S of integers in unary is understood as a predicate
+ * that evaluates to true on the elements of the set.
+ *
+ * context.binary contains constraints over pairs of states that are
+ * layered by time frames. Every set R of integer pairs in binary is
+ * understood as a predicate that evaluates to true on the elements of the
+ * set.
+ *
+ * @type: (CONTEXT, Seq(Int)) => Bool;
+ *)
+IsModel(ctx, trace) ==
+    /\  \A i \in DOMAIN trace:
+          \A Pred \in ctx.unary[i]:
+            trace[i] \in Pred
```

Comment:
Why this complicated definition if we only every push `{}` into the context after the initial states?